src/main/scala/pureconfig/ConfigConvert.scala

```diff
@@ -277,32 +283,16 @@ object ConfigConvert extends LowPriorityConfigConvertImplicits {
  * Implicit [[ConfigConvert]] instances defined such that they can be overriden by library consumer via a locally defined implementation.
  */
 trait LowPriorityConfigConvertImplicits {
-  import scala.concurrent.duration.{ Duration, FiniteDuration }
-  implicit val durationConfigConvert: ConfigConvert[Duration] = new ConfigConvert[Duration] {
-    override def from(config: ConfigValue): Try[Duration] = config match {
-      case conf @ (_: ConfigObject | _: ConfigList) => Failure(WrongTypeException(conf.valueType().toString))
-      case null => Failure(CannotConvertNullException)
-      case other => Some(other.render(ConfigRenderOptions.concise())).fold[Try[Duration]](Failure(new IllegalArgumentException(s"Couldn't read duration from $config."))) { durationString =>
-        DurationConvert.from(durationString).recoverWith {
-          case ex => Failure(new IllegalArgumentException(s"Could not parse a duration from '$durationString'. (try ns, us, ms, s, m, h, d)"))
-        }
-      }
-    }
-    override def to(t: Duration): ConfigValue = {
-      ConfigValueFactory.fromAnyRef(DurationConvert.from(t))
-    }
+  implicit val durationConfigConvert: ConfigConvert[Duration] = {
+    nonEmptyStringConvert(s => DurationConvert.fromString(s, implicitly[ClassTag[Duration]]).get, DurationConvert.fromDuration)
   }
 
-  implicit val finiteDurationConfigConvert: ConfigConvert[FiniteDuration] = new ConfigConvert[FiniteDuration] {
-    override def from(config: ConfigValue): Try[FiniteDuration] = durationConfigConvert.from(config) match {
-      case Success(v) =>
-        if (v.isFinite())
-          Success(Duration(v.length, v.unit))
-        else
-          Failure(new IllegalArgumentException(s"Couldn't parse a finite duration from a duration that is not finite."))
-      case Failure(f) => Failure(f)
+  implicit val finiteDurationConfigConvert: ConfigConvert[FiniteDuration] = {
+    val fromString: String => FiniteDuration = (s: String) => DurationConvert.fromString(s, implicitly[ClassTag[FiniteDuration]]).get match {
```

Comment:
The signature of `nonEmptyStringConvert` means I have to throw on failure, right? (I hope I'm not misunderstanding you.)

I could rewrite it as:
```
  implicit val finiteDurationConfigConvert: ConfigConvert[FiniteDuration] = {
    val fromString: String => FiniteDuration = { (s: String) =>
      DurationConvert.fromString(s, implicitly[ClassTag[FiniteDuration]])
        .flatMap { d =>
          if (d.isFinite()) Success(Duration(d.length, d.unit))
          else Failure(new IllegalArgumentException(s"Couldn't parse '$s' into a finite duration because it's infinite."))
        }
        .get
    }
    nonEmptyStringConvert(fromString, DurationConvert.fromDuration)
  }
```

Is that what you were thinking.